src/components/App/ExpressRecorder.tsx

```diff
@@ -486,27 +502,49 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
                 });
                 return;
             }
-            if (screenOn && screenOn !== shareScreenOn) {
+            if (screenOn && (screenOn !== shareScreenOn || !finalConstraints.video)) {
+                if (!finalConstraints.video && finalConstraints.audio) {
```

Comment:
it's because the functionallity is complicated...
I need to conisider all combinations.
the way you seperated the condition is not equivalent to what I need.

Reply:
of course, it's not.
I wanna show you the agenda and not write your code.

Reply:
yes. and you can even destruct the `finalConstraints` object for better vars isolating. 
```
const { video, audio } = finalConstraints;
```

Reply:
done